src/BlockchainReader/Bitcoin.ts

```diff
@@ -19,7 +33,7 @@ export const blockToPoetAnchors = (block: Block): ReadonlyArray<PoetBlockAnchor>
 const dataOutputToPoetTransactionAnchors = (acc: ReadonlyArray<PoetTransactionAnchor>, dataOutput: VOutWithTxId) => {
   const buffer = dataOutputToBuffer(dataOutput)
   const poetAnchor = bufferToPoetAnchor(buffer)
-  return [...acc, combineAnchorAndTransactionId(poetAnchor, dataOutput)]
+  return isCorrectBufferLength(buffer) ? [...acc, combineAnchorAndTransactionId(poetAnchor, dataOutput)] : acc
```

Comment:
```suggestion
  return isCorrectBufferLength(buffer) ? [...acc, combineAnchorAndTransactionId(bufferToPoetAnchor(buffer), dataOutput)] : acc
```

might move the bufferToPoetAnchor call to the return statement so that we are not creating a poetAnchor from an incorrect buffer.